pkg/api/controllers/volume.go

```diff
@@ -30,21 +30,23 @@ import (
 	c "github.com/opensds/opensds/pkg/context"
 	"github.com/opensds/opensds/pkg/controller/client"
 	"github.com/opensds/opensds/pkg/db"
+	dock "github.com/opensds/opensds/pkg/dock/client"
 	"github.com/opensds/opensds/pkg/model"
 	pb "github.com/opensds/opensds/pkg/model/proto"
 	. "github.com/opensds/opensds/pkg/utils/config"
 )
 
 func NewVolumePortal() *VolumePortal {
 	return &VolumePortal{
-		CtrClient: client.NewClient(),
+		CtrClient:  client.NewClient(),
+		DockClient: dock.NewClient(),
 	}
 }
 
 type VolumePortal struct {
 	BasePortal
-
-	CtrClient client.Client
+	CtrClient  client.Client
+	DockClient dock.Client
```

Comment:
If the idea works well, then DockClient field can be removed, and a lot of code can be reduced.